crates/core/src/factories/core_fetch.rs

```diff
@@ -146,6 +165,38 @@ impl FetchFactory for CoreFetchFactory {
 }
 
 type FetchRequest = (OpId, AgentId);
+type FetchResponse = (Vec<OpId>, Url);
+
+#[derive(Debug)]
+struct FetchResponseHandler {
+    response_tx: Sender<FetchResponse>,
+}
+
+impl TxModuleHandler for FetchResponseHandler {
+    fn recv_module_msg(
+        &self,
+        peer: Url,
+        _space: SpaceId,
+        _module: String,
+        data: bytes::Bytes,
+    ) -> K2Result<()> {
+        let op_ids = deserialize_op_ids(data)?;
+        if let Err(err) = self.response_tx.try_send((op_ids, peer)) {
+            tracing::warn!(
+                "could not pass fetch request to response task: {err}",
+            );
+        };
+        Ok(())
+    }
+}
+
+impl TxBaseHandler for FetchResponseHandler {
+    fn peer_connect(&self, _peer: Url) -> K2Result<()> {
+        Ok(())
+    }
+    fn new_listening_address(&self, _this_url: Url) {}
+    fn peer_disconnect(&self, _peer: Url, _reason: Option<String>) {}
+}
```

Comment:
pretty sure these all have default impls

```suggestion
impl TxBaseHandler for FetchResponseHandler {}
```

Reply:
That hadn't occurred to me.